### ISSUE TITLE
`no-navigation-without-resolve` no error on page data key-value

### ISSUE THREAD

**Issue by hyunbinseo** [sveltejs/eslint-plugin-svelte#1526]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

10.4.0

### What version of `eslint-plugin-svelte` are you using?

3.17.1

### What did you do?

No lint error on `goto(data.path);` part.

```svelte
<!-- +page.svelte -->
<script lang="ts">
  import { goto } from '$app/navigation';

  const { data } = $props();

  (() => {
    // (property) path: string
    goto(data.path);
  })();
</script>
```

```ts
// +page.ts
export const load = () => ({ path: '' });
```

### What did you expect to happen?

```ts
goto(data.path); // Unexpected goto() call without resolve().
```

### What actually happened?

```ts
goto(data.path); // no error
```

### Link to **GitHub Repo** with Minimal Reproducible Example

https://github.com/hyunbinseo/svelte-kit-templates/tree/eslint-plugin-svelte-1526

### Additional comments

Seems to be broken from `eslint-plugin-svelte@3.17.1`. Downgrading to `3.17.0` shows the error.

**Comment by marekdedic** [comment]
I think the fact that `path` is a string isn't "seen" by eslint (as it works per-file). So it is typed as `any`. If you replace the props declaration by this

```ts
const { data }: { data: { path: string } } = $props();
```
then it (correctly) reports an error.

I think we can think about whether `any` should trigger an error (probably yes).

**Comment by hyunbinseo** [comment]
Upgrading to `3.17.1` will be breaking changes for people since ESLint errors will be shown:

```
  91:5  warning  Unused eslint-disable directive (no problems were reported from 'svelte/no-navigation-without-resolve')
  118:4  warning  Unused eslint-disable directive (no problems were reported from 'svelte/no-navigation-without-resolve')
```

Should I remove the `eslint-disable` comments for now until this is possibly resolved?

**Comment by marekdedic** [comment]
Ok, then let's talk about typing and `any` :). Do we think that this should trigger the rule?

```svelte
<script lang="ts">
  import { goto } from '$app/navigation';

  const x = "/" as any;
  goto(x);
</script>
```

I think that it should... @ota-meshi thoughts?

**Comment by ota-meshi** [comment]
Since the `any` type can accept the `ResolvedPathname` type, I think it's correct for the rule not to report it, considering the consistency of handling.

**Comment by marekdedic** [comment]
That actually makes sense, agree. I can imagine in the future adding a configuration option to the rule (something like `reportAny`, default `false`), but that's lower in my priorities.
